### PR TITLE
Install gloo-fed alongside gloo-ee

### PR DESCRIPTION
**Pull request by npolshakova**
# Description

Gloo-fed will now be installed along side with enterprise with: `glooctl install enterprise` by default. The `--with-gloo-fed=[true/false]` flag can disable installing enterprise with gloo-fed. 

Separate files for gloo-fed and gloo can be specified when installing enterprise: `--file [local-gloo-ee-file]  --gloo-fed-file [local-gloo-fed-file]`

Separate value files to override the helm charts for gloo-fed and gloo can be specified with: `--values [list of gloo values] --gloo-fed-values [list of gloo-fed value files]`

Separate release names can be specified with: `--release-name [release name for gloo] --gloo-fed-release-name [release name for gloo-fed]`

Separate namespaces can be specified with: `--namespace [gloo namespace] --gloo-fed-namespace [gloo-fed ns overwrite]`

# Context

Part of https://github.com/solo-io/gloo/issues/4267

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ashleywang1** [comment]
Another test that would be useful for this is to pull in this branch into solo-projects, and modify the Gloo Fed e2e test to run this glooctl command (`glooctl install enterprise --file [local-gloo-ee-file] --with-gloo-fed --gloo-fed-file (?)]

File: projects/gloo/cli/pkg/cmd/install/enterprise.go [hunk 1]
```diff
@@ -45,5 +45,10 @@ func enterpriseCmd(opts *options.Options) *cobra.Command {
 	pFlags := cmd.PersistentFlags()
 	flagutils.AddGlooInstallFlags(cmd.Flags(), &opts.Install)
 	flagutils.AddEnterpriseInstallFlags(pFlags, &opts.Install)
+	pFlags.Lookup("gloo-fed-file").Hidden = true
+	pFlags.Lookup("gloo-fed-values").Hidden = true
+	pFlags.Lookup("gloo-fed-release-name").Hidden = true
+	pFlags.Lookup("gloo-fed-create-namespace").Hidden = true
+	pFlags.Lookup("gloo-fed-namespace").Hidden = true
```
Comment (ashleywang1) [inline projects/gloo/cli/pkg/cmd/install/enterprise.go:52]: 👍 nice! We can unhide these values later if users ask for them!

**Comment by ashleywang1** [comment]
This LGTM! Waiting for test to pass + running a local install

File: projects/gloo/cli/pkg/cmd/install/install_test.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ import (
 var _ = Describe("Install", func() {
 
 	const licenseKey = "--license-key=fake-license-key"
-	const overrideVersion = "0.20.7"
+	const overrideVersion = "1.7.0-beta11"
```
Comment (npolshakova) [inline projects/gloo/cli/pkg/cmd/install/install_test.go:18]: Now that gloo-fed and gloo use the same version, need an override version that both gloo-fed and gloo have a helm chart for.

**Comment by ashleywang1** [comment]
Final notes:
```
# To uninstall everything
glooctl uninstall --all

# just uninstalls Gloo EE
glooctl uninstall gateway enterprise --all

# just uninstalls Gloo Fed 
glooctl uninstall federation --all

**Review by ashleywang1 — APPROVED** [review]

**Pull request merged**